/home/jules/.pyenv/versions/3.12.12/lib/python3.12/site-packages/pygame/pkgdata.py:25: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  from pkg_resources import resource_stream, resource_exists
error: XDG_RUNTIME_DIR is invalid or not set in the environment.
ALSA lib confmisc.c:855:(parse_card) cannot find card '0'
ALSA lib conf.c:5204:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory
ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings
ALSA lib conf.c:5204:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1342:(snd_func_refer) error evaluating name
ALSA lib conf.c:5204:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5727:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2721:(snd_pcm_open_noupdate) Unknown PCM default
pygame 2.6.1 (SDL 2.28.4, Python 3.12.12)
Hello from the pygame community. https://www.pygame.org/contribute.html
Testing model: final_race_model.pth

Episode 1
Caught a TypeError: points must be number pairs
DEBUG: car_corners in test_model = [(np.float32(656.01904), np.float32(290.3118)), (np.float32(655.97974), np.float32(310.31174)), (np.float32(643.97974), np.float32(310.28818)), (np.float32(644.01904), np.float32(290.28824))]
Traceback (most recent call last):
  File "/app/train_model.py", line 356, in <module>
    test_model(model_path=args.model)
  File "/app/train_model.py", line 323, in test_model
    raise e
  File "/app/train_model.py", line 318, in test_model
    env.render()
  File "/app/race_environment.py", line 588, in render
    pygame.draw.polygon(self.screen, car_color, car_corners)
TypeError: points must be number pairs
